### PR TITLE
Update symbolics policy to emit aten::ATen for Caffe2 build only

### LINKED ISSUE #75623: [ONNX] `ONNX_ATEN_FALLBACK` supports ONNX Runtime
Previously setting `operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK` produed a model that was only usable by Caffe2.
This issue tracks work to make it usable by ONNX Runtime as well.

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Currently ONNX exporter symbolics can emit ATen operators when `operator_export_type==ONNX_ATEN_FALLBACK`. However, this is a behavior specific to Caffe2 builds, as the intend use of `ONNX_ATEN_FALLBACK` is to emit ATen operators only when there is no ONNX equivalent.

The reason Caffe2 choses to emit ATen operators when ONNX counterpart exists is for performance on their particular engine implementation, which might not be true for other implementations. e.g. ONNX Runtime can optimize the generated ONNX graph into something more efficient

This PR must be merged only after https://github.com/pytorch/pytorch/pull/73954

### CODE REVIEW

File: test/jit/test_export_modes.py [hunk 1]
```diff
@@ -68,6 +68,24 @@ def foo(a):
         x = torch.ones(3)
         torch.onnx._export(foo, (x,), f)
 
+    @skipIfNoCaffe2
+    @skipIfNoLapack
+    def test_caffe2_aten_fallback(self):
```
Comment (garymm) [inline test/jit/test_export_modes.py:73]: Can this go in test_pytorch_onnx_caffe2.py instead?

File: test/onnx/test_operators.py [hunk 2]
```diff
@@ -593,7 +593,7 @@ def test_batchnorm_noaffine(self):
         self.assertONNX(nn.BatchNorm2d(128, affine=False, momentum=0.3), x,
                         keep_initializers_as_inputs=True)
 
-    @skipIfCaffe2
+    @skipIfNoCaffe2
```
Comment (garymm) [inline test/onnx/test_operators.py:596]: I think tests in this file should test the non-Caffe2 specific behavior whenever possible, as that's what will affect most users going forard. So I think the old decorator was more appropriate. WDYT?

File: test/onnx/test_operators.py [hunk 3]
```diff
@@ -793,7 +793,7 @@ def forward(self, input, input2):
         input2 = torch.arange(24, dtype=torch.uint8).reshape(3, 4, 2)
         self.assertONNX(BitshiftModel(), (input, input2), opset_version=11)
 
-    @skipIfCaffe2
+    @skipIfNoCaffe2
```
Comment (garymm) [inline test/onnx/test_operators.py:796]: ditto

**Review by garymm — COMMENTED** [review]
Thanks!

File: test/onnx/test_operators.py (hunk 2 above)
Comment (thiagocrepaldi) [reply test/onnx/test_operators.py:596]: The test was originally added for Caffe2, so I kept this behavior.
I am OK with converting to aten-non-caffe2 or even have both caffe2 and non-caffe2 versions

Comment (garymm) [reply test/onnx/test_operators.py:596]: I don't understand how the test could originally be for caffe2 if it was decorated with `skipIfCaffe2`?
Please explain.

Comment (thiagocrepaldi) [reply test/onnx/test_operators.py:596]: I did that in the past https://github.com/pytorch/pytorch/commit/9bbe1d632e6536934550b6973366e2acdd4e1995 and now plan to restore original behavior and guard legacy Caffe2 from our changes

File: test/jit/test_export_modes.py (hunk 1 above)
Comment (thiagocrepaldi) [reply test/jit/test_export_modes.py:73]: This is a `export mode` test file, which includes caffe2 as one of the modes. Are you sure we should move it?

File: test/onnx/test_operators.py (hunk 2 above)
Comment (garymm) [reply test/onnx/test_operators.py:596]: My vote is to either delete it or keep it as a non-Caffe2 test.

File: test/jit/test_export_modes.py (hunk 1 above)
Comment (garymm) [reply test/jit/test_export_modes.py:73]: This doesn't have AssertONNX either?
In any case, I can see why this test makes sense to keep in this file, so I withdraw my request. This is OK to keep here.

File: test/onnx/test_operators.py (hunk 2 above)
Comment (thiagocrepaldi) [reply test/onnx/test_operators.py]: ok, changing to non-caffe2 test

File: test/onnx/test_operators.py (hunk 3 above)
Comment (thiagocrepaldi) [reply test/onnx/test_operators.py]: changed to non-caffe2 test

File: test/jit/test_export_modes.py (hunk 1 above)
Comment (thiagocrepaldi) [reply test/jit/test_export_modes.py:73]: kept here

**Review by garymm — APPROVED** [review]
⛈️

**Comment by thiagocrepaldi** [comment]
@glaringlee @anjali411 @simpkins @bugra @dzhulgakov adding you as we need to `test/jit/test_export_modes.py` onnx test in the `merge_rules.json` - or we can move it to onnx test folder (from jit) with your approval

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**